src/App.js

```diff
@@ -14,11 +14,111 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    title: 'No goods selected',
+    // eslint-disable-next-line react/no-unused-state
+    list: goodsFromServer,
+    selectedGood: [],
+  };
+
+  onAddProduct = (newTitle) => {
+    this.state.selectedGood.push(newTitle);
+    this.setState({
+      title: `${newTitle} is selected`,
+    });
+  };
+
+  onDeleteProduct = (product) => {
+    const index = this.state.selectedGood.indexOf(product);
+
+    if (index > -1) {
+      const { selectedGood: prev } = this.state;
+
+      prev.splice(index, 1);
+
+      this.setState({
+        selectedGood: prev,
+        title: `${product} is deleted`,
+      });
+    } else {
+      this.setState({
+        title: `${product} was not selected`,
+      });
+    }
+  };
+
+  clear = () => {
+    this.setState({
+      selectedGood: [],
+      title: 'No goods selected',
+    });
+  };
+
+  render() {
+    return (
+      <div className="wrapper">
+        <h1>{this.state.title}</h1>
+        <button
+          type="button"
+          className={this.state.selectedGood.length === 0 ? 'visibility' : null}
+          onClick={this.clear}
+        >
+          X
+        </button>
+        <p>
+          <strong>Your list: </strong>
+          {this.state.selectedGood.join(', ')}
+        </p>
+        <div
+          style={{
+            width: 700,
+            border: '1px solid #ccc',
+            backgroundColor: 'beige',
+            margin: '20px auto',
+          }}
+        >
+          {this.state.list.map(product => (
+            <>
+              <ul className="list" key={product}>
+                <li
```

Comment:
```suggestion
          <ul className="list">
            {this.state.list.map(product => (
              <li key={product}
```
You should render (map) list items inside `ul` tag. Now you have a lot of lists (`ul`). There should be one `ul` with many `li` inside.
Also, `key` attribute should be applied to `li` (not `ul`)